scripts/linting/mdn-sitemap-compare.js

```diff
@@ -0,0 +1,130 @@
+/*
+A linter that can check that all of our URLs match URLs in MDN's sitemaps.
+
+Usage:
+
+npm run mdn-sitemap-compare [locale [locale2]]
+
+*/
+
+const url = require("url");
+const fetch = require("node-fetch");
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+
+const readFile = util.promisify(fs.readFile);
+
+const baseURL = "https://developer.mozilla.org";
+
+const ROOT = path.join(__dirname, "..", "..");
+
+async function fetchSitemap(url) {
+    if (url.startsWith("/")) url = `${baseURL}${url}`;
+    let r = await fetch(url);
+    let payload = await r.text();
+    return payload;
+}
+
+async function getLocs(url) {
+    const payload = await fetchSitemap(url);
+    return [...payload.matchAll(/<loc>(.*?)<\/loc>/g)].map(m => m[1]);
+}
+
+async function getMDNURL(filepath) {
+    const content = await readFile(filepath, "utf8");
+    const match = content.match(/mdn_url:\s*(.*)/);
```

Comment:
I think this would be safer and clearer if it explicitly loaded the front matter:

```js
const matter = require('gray-matter');
const { data, content } = matter(docs);
```